CMakeLists.txt

```
@@ -431,8 +431,31 @@ endif()
 
 if(DEFINED BUILD_VERSION)
   set(build_version_argument "-DBUILD_VERSION=${BUILD_VERSION}")
-elseif(EXISTS ${ZEPHYR_BASE}/.git/index)
+elseif(IS_DIRECTORY ${ZEPHYR_BASE}/.git)
   set(git_dependency ${ZEPHYR_BASE}/.git/index)
+elseif(EXISTS ${ZEPHYR_BASE}/.git)
+  # Likely a git-submodule. Let's ask git where the real database is located.
+  find_package(Git QUIET)
+  if(GIT_FOUND)
+    execute_process(
+      COMMAND ${GIT_EXECUTABLE} rev-parse --git-dir
```

`--git-dir` is not guaranteed to output an absolute path, although the current behaviour seems to be to output absolute path when the working directory does not contain the `.git` directory.

```suggestion
      COMMAND ${GIT_EXECUTABLE} rev-parse --absolute-git-dir
```

---

Interesting. Always thought `--git-dir` did what `--absolute-git-dir` does. I stand corrected.